src/client/eppo-precomputed-client.ts

```diff
@@ -43,44 +50,51 @@ export type PrecomputedFlagsRequestParameters = {
   skipInitialPoll?: boolean;
 };
 
+interface EppoPrecomputedClientOptions {
+  precomputedFlagStore: IConfigurationStore<PrecomputedFlag>;
+  subject: Subject;
+  requestParameters?: PrecomputedFlagsRequestParameters;
+}
+
 export default class EppoPrecomputedClient {
   private readonly queuedAssignmentEvents: IAssignmentEvent[] = [];
   private assignmentLogger?: IAssignmentLogger;
   private assignmentCache?: AssignmentCache;
   private requestPoller?: IPoller;
-  private precomputedFlagsRequestParameters?: PrecomputedFlagsRequestParameters;
-  private subjectKey?: string;
-  private subjectAttributes?: Attributes;
-
-  constructor(
-    private precomputedFlagStore: IConfigurationStore<PrecomputedFlag>,
-    private isObfuscated = false,
-  ) {}
-
-  public setPrecomputedFlagsRequestParameters(
-    precomputedFlagsRequestParameters: PrecomputedFlagsRequestParameters,
-  ) {
-    this.precomputedFlagsRequestParameters = precomputedFlagsRequestParameters;
-  }
-
-  public setSubjectAndPrecomputedFlagsRequestParameters(
-    precomputedFlagsRequestParameters: PrecomputedFlagsRequestParameters,
-  ) {
-    this.setPrecomputedFlagsRequestParameters(precomputedFlagsRequestParameters);
-    this.subjectKey = precomputedFlagsRequestParameters.precompute.subjectKey;
-    this.subjectAttributes = precomputedFlagsRequestParameters.precompute.subjectAttributes;
-  }
-
-  public setPrecomputedFlagStore(precomputedFlagStore: IConfigurationStore<PrecomputedFlag>) {
-    this.precomputedFlagStore = precomputedFlagStore;
-  }
+  private requestParameters?: PrecomputedFlagsRequestParameters;
+  private subject: {
+    subjectKey: string;
+    subjectAttributes: ContextAttributes;
+  };
+  private precomputedFlagStore: IConfigurationStore<PrecomputedFlag>;
 
-  public setIsObfuscated(isObfuscated: boolean) {
-    this.isObfuscated = isObfuscated;
+  public constructor(options: EppoPrecomputedClientOptions) {
+    this.precomputedFlagStore = options.precomputedFlagStore;
+    const { subjectKey, subjectAttributes } = options.subject;
+    this.subject = {
+      subjectKey,
+      subjectAttributes: ensureContextualSubjectAttributes(subjectAttributes),
+    };
+    if (options.requestParameters) {
+      // Online-mode
+      this.requestParameters = options.requestParameters;
+    } else {
+      // Offline-mode
+      if (!this.precomputedFlagStore.isInitialized()) {
+        logger.error(
+          '[Eppo SDK] EppoPrecomputedClient requires an initialized precomputedFlagStore if requestParameters are not provided',
+        );
+      }
+      if (!this.precomputedFlagStore.salt) {
+        logger.error(
+          '[Eppo SDK] EppoPrecomputedClient requires a precomputedFlagStore with a salt if requestParameters are not provided',
+        );
+      }
```

Comment:
Printing instead of throwing errors if the flag store is misconfigured because the SDK shouldn't disrupt the user's code, but also these scenarios should never happen, and if they do, assignments will always return the user-provided default